src/ComunicaEngine.js

```diff
@@ -31,13 +36,21 @@ export default class ComunicaEngine {
     let bindings;
     const next = async () => {
       if (!bindings) {
-        // Determine the document to query from the subject if there is no source
-        const source = this._source ?
-          { type: 'rdfjsSource', value: this._source } :
-          { type: 'file', value: this.getDocument(await this._subject) };
+        let sources;
+        if (this._source) {
+          const value = await this._source;
```

Comment:
`const source`

Reply:
Named it value so it would auto-expand in the final source object where `value` is the the name of the url field. ;) But I'll change it then.

Reply:
ah I missed that